mypyc/ops.py

```diff
@@ -1174,41 +1190,38 @@ def __str__(self) -> str:
         return '\n'.join(format_func(self))
 
 
+VTableMethod = NamedTuple(
+    'VTableMethod', [('cls', 'ClassIR'),
+                     ('method', FuncIR)])
+
+
+VTableAttr = NamedTuple(
+    'VTableAttr', [('cls', 'ClassIR'),
+                   ('name', str),
+                   ('is_getter', bool)])
+
+
 class ClassIR:
     """Intermediate representation of a class.
 
     This also describes the runtime structure of native instances.
     """
 
-    # TODO: Use dictionary for attributes in addition to (or instead of) list.
-
     def __init__(self, name: str, module_name: str) -> None:
         self.name = name
         self.module_name = module_name
         self.attributes = OrderedDict()  # type: OrderedDict[str, RType]
-        self.methods = []  # type: List[FuncIR]
+        self.method_types = OrderedDict()  # type: OrderedDict[str, FuncSignature]
```

Comment:
Why is this separate from `methods`, even though `FunciR` has a signature? If the signature here can be different from the one in `methods`, add a comment describing this.

Reply:
I'll add a comment explaining, but it's because we populate the method_types before we populate the methods, and we rely on them being present in some places.